code/game/machinery/vending.dm

```diff
@@ -708,7 +708,7 @@
 /obj/machinery/vending/proc/throw_item()
 	var/obj/throw_item = null
 	var/mob/living/target = locate() in view(7,src)
-	if(!target)
+	if(!target || !target.client)

```

Comment:
Shouldn't this try to find another valid target if there is one instead of just `return`ing?

Reply:
throw the ssd infront of rogue machines to make them stop firing!

Reply:
@farie82 it would only stop firing if it was already targeting the SSD player

Reply:
@Citinited also that opens the possibility of endless loops

Reply:
This means monkies and mobs won't be hit by this anymore.

I don't like that.

Is this *really* that much of a problem?

Reply:
yeah it spams the logs a whole lot. I'll just switch to a player_logged check then